### PR TITLE
Decoding analysis to make a resampling distribution

### PR DESCRIPTION
**Pull request by resatomo**
This function can be done repeatedly with different random seeds to get a resampling distribution. It can be used with multiprocessing to make the computation faster.

### CODE REVIEW

File: aopy/analysis/base.py [hunk 1]
```diff
@@ -2316,7 +2317,159 @@ def windowed_xval_lda_wrapper(data, labels, samplerate, lags=3, nfolds=5, regula
...
+                                Can either be None, 'auto', or a float between 0 and 1. 
+        lda_model (sklearn LDA class, optional): User-defined LDA model from sklearn.discriminant_analysis.LinearDiscriminantAnalysis. If None, this function will initialize the model.
+        return_weights (bool, optional): Whether to return the LDA weights (default is False).
+        seed (int, optional): random seed
+
+    Returns:
+        tuple: Tuple containing:
+            | **pred_Y (nch,ntr) / (ntr):** a list of predicted labels. If single_decoding is False, its shape becomes (ntr).
+            | **true_Y (nch,ntr) / (ntr):** a list of true labels. If single_decoding is False, its shape becomes (ntr).
+            | **(Optional) LDA Weights (nlabels, nch, nfolds):** The LDA weights if return_weights is True.
```
Comment (rcanfield2) [inline aopy/analysis/base.py:2353]: I think this function should compute the decoding accuracy and return it. Returning the labels can be and option

File: aopy/analysis/base.py [hunk 2]
```diff
@@ -2316,7 +2317,159 @@ def windowed_xval_lda_wrapper(data, labels, samplerate, lags=3, nfolds=5, regula
...
+        data (nunit, ntr): neural data
+        labels (ntr): the labels for each trial.
+        min_trial (int): the minimum number of trials for each label to be extracted
+                                if this is 20, 20*(the number of unique labels) trials are extracted
+        cond_mask (ntr): a boolean array. This is a mask to extract trials that satisfies a certain condition
+                                (ex. movement onset is more than a certain threshold)
+        single_decoding (bool, optional): If True, LDA is performed using single unit activity separately (default is True)
+        min_unit (int, optional): the number of units used for decoding. This is used only when single_decoding is False.
+                                if min_unit is None, decoding is perfomed using all units (default is None)
+        shuffle (bool, optional): whether to shuffle labels or not (default is False)
```
Comment (rcanfield2) [inline aopy/analysis/base.py:2339]: Can you rename this to 'shuffle_labels'? To be more specific that units or trials are not shuffled.

**Review by rcanfield2 — COMMENTED** [review]
Looks good, just some documentation comments/questions.

**Comment by resatomo** [comment]
I also modified code to use multiple features such as different band powers.

**Review by rcanfield2 — APPROVED** [review]
